test/read_test.py

```diff
@@ -169,26 +128,25 @@ def embiggen(e):
 
 @pytest.mark.asyncio
 async def test_iter_from_missing_stream(event_loop):
-
-    async with connect(loop=event_loop) as c:
-        try:
+    async with connect(loop=event_loop, username="test-user", password="test-password") as c:
+        with pytest.raises(exceptions.StreamNotFound):
             [e async for e in c.iter("my-stream-that-isnt-a-stream")]
-            assert False
-        except Exception as e:
-            assert isinstance(e, exceptions.StreamNotFound)
 
 
 @pytest.mark.asyncio
 async def test_iterall(event_loop):
     async with connect(
-        loop=event_loop, name="iter_all", username="admin", password="changeit"
+        loop=event_loop,
+        username="admin",  # iter_all aggregates all streams so it needs systemwide-read perms
+        password="changeit",
+        name="iter_all",
     ) as c:
         stream_name = str(uuid.uuid4())
-        await given_a_stream_with_three_events(c, stream_name)
+        await given_two_streams_with_two_events(c, stream_name)
```

Comment:
this tests checks an API that lets you read from _all_ streams.

i looked at the commit history and found that when this test was first written it used this `given_two_streams...` fixture, but a later refactor (i think accidentally) changed it to using a single-stream fixture.

i think changing it back makes for a better test.